GVRf/Framework/framework/src/main/java/org/gearvrf/script/GVRContextProxy.java

```diff
@@ -0,0 +1,352 @@
+package org.gearvrf.script;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+
+import org.gearvrf.GVRActivity;
+import org.gearvrf.GVRAndroidResource;
+import org.gearvrf.GVRAssetLoader;
+import org.gearvrf.GVRAtlasInformation;
+import org.gearvrf.GVRBitmapTexture;
+import org.gearvrf.GVRContext;
+import org.gearvrf.GVRCubemapTexture;
+import org.gearvrf.GVRDrawFrameListener;
+import org.gearvrf.GVREventManager;
+import org.gearvrf.GVREventReceiver;
+import org.gearvrf.GVRImportSettings;
+import org.gearvrf.GVRMaterialShaderManager;
+import org.gearvrf.GVRMesh;
+import org.gearvrf.GVRPostEffectShaderManager;
+import org.gearvrf.GVRScene;
+import org.gearvrf.GVRSceneObject;
+import org.gearvrf.GVRScreenshot3DCallback;
+import org.gearvrf.GVRScreenshotCallback;
+import org.gearvrf.GVRTexture;
+import org.gearvrf.GVRTextureParameters;
+import org.gearvrf.animation.GVRAnimationEngine;
+import org.gearvrf.debug.DebugServer;
+import org.gearvrf.io.GVRInputManager;
+import org.gearvrf.periodic.GVRPeriodicEngine;
+import org.gearvrf.scene_objects.GVRModelSceneObject;
+
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.concurrent.Future;
+
+public final class GVRContextProxy {
+    private final WeakReference<GVRContext> mContext;
```

Comment:
Technically, we should. In this case though I consider it invalid state. If a script tries to use the "gvrf" global to access an activity that has been gc-ed then I rather have an exception thrown since it is not supposed to happen.